back-end/helpers/get/transcript_data_grouper.spec.js

```diff
@@ -0,0 +1,417 @@
+                intents.push({
+                    ...fake,
+                    children: new Map(fake.children),
+                });
+            }
+            jest.spyOn(fakeDao, "getIntent").mockImplementation((query) => {
+                const result = intents.filter((intent) => {
+                    return intent.question === query.question && intent.project_id === query.project_id;
+                });
+                const newChildren = new Map(JSON.parse(JSON.stringify(Array.from(result[0].children))))
+                const deepIntent = JSON.parse(JSON.stringify(result[0]))
+                deepIntent.children = newChildren
+                return [deepIntent];
+            });
+        });
+
+        it("should correctly merge 2 similar intents", async() => {
+            const alreadyProcessedIntents = intents.slice(0, 1);
+            const similarIntent = intents[1];
+            const grouper = new TranscriptDataGrouper(alreadyProcessedIntents, 1);
+            
+            const result = await grouper.group(similarIntent);
+            expect(fakeDao.getIntent).toHaveBeenCalled();
+            expect(intents[0].children["h"]).toEqual(33);
+            expect(intents[0].children["c"]).toEqual(33);
+            expect(intents[0].children["b"]).toEqual(33);
+            expect(result).toBe(true);
+        });
+
+        it("should correctly merge 3 similar intents", async() => {
+            /* TODO: Test is currently broken
+            / Currently receiving false but need to check reasons for why it is not computing properly
+            / Probably in recalculation step where children is not defined
+            */
+
+            const alreadyProcessedIntents = intents.slice(0, 1);
+            const grouper = new TranscriptDataGrouper(alreadyProcessedIntents, 1);
+
+            for (const similarIntent of intents.slice(1, 3)) {
+                const result = await grouper.group(similarIntent);
+                expect(fakeDao.getIntent).toHaveBeenCalled();
+                expect(result).toBe(true);
+            }
+            expect(intents[0].children["h"]).toEqual(33);
+            expect(intents[0].children["c"]).toEqual(22);
+            expect(intents[0].children["b"]).toEqual(44);
+        });
+
+        it("should correctly merge first 3 similar intents and not the rest", async() => {
+            // TODO: Test is also broken
+
+            const alreadyProcessedIntents = intents.slice(0, 1);
+            const grouper = new TranscriptDataGrouper(alreadyProcessedIntents, 1);
+
+            for (const intent of intents.slice(1)) {
+                await grouper.group(intent);
+                expect(fakeDao.getIntent).toHaveBeenCalled();
+            }
+            expect(intents[0].children["h"]).toEqual(33);
+            expect(intents[0].children["c"]).toEqual(22);
+            expect(intents[0].children["b"]).toEqual(44);
+        });
+    });
+    describe("when intents are similar but separated", () => {
```

Comment:
Can you explain why this is a separate section. I don't really understand why it is significantly different from the previous block?

Reply:
I wanted to separate it so that a later intent will be merged with an earlier one even with dissimilar intents in between

Reply:
I think it would be better to rearrange the 2nd baseIntent array than declare a new one then since it adds a lot of bulk to the test file.